src/main/java/software/amazon/msk/auth/iam/internals/MSKCredentialProvider.java

```diff
@@ -261,6 +273,10 @@ public List<AwsCredentialsProvider> getProviders() {
             return providers;
         }
 
+        public Boolean addDefaultProviders() {
+            return Optional.ofNullable(optionsMap.get(AWS_ADD_DEFAULT_PROVIDERS)).map(d -> d.equals("true")).orElse(true);
```

Comment:
It might be better to check if the value is (not) `false` instead to better align with how this will work.

```suggestion
            return Optional.ofNullable(optionsMap.get(AWS_ADD_DEFAULT_PROVIDERS)).map(d -> !d.equals("false")).orElse(true);
```

Reply:
I don't think we need to overprotect backward compatibility. if the parameter is not provided, the default value is `true`, to provide full backward compatibility, but if param is provided, let's treat it as the rest of the booleans for debug and fips. first, to be consistent, second, it's always boolean in most cases resolves to false, unless explicitly defined as `true`